README.md

```diff
@@ -538,6 +541,54 @@ ISomeApi = RestClient.For<ISomeApi>("http://api.example.com");
 await api.FooAsync("bar/baz");
 ```
 
+#### Serialization of Path Parameters
+
+Similar to query paramters, calling `ToString()` is sometimes not enough: you might want to apply custom conversion to some parameters where `ToString()` is not feasible (like enum values).
+In this case, you can mark the parameter for custom serialization using `PathSerializationMethod.Serialized`, and further control it by using a [custom serializer](#custom-serializers-and-deserializers).
+
+For example:
+```csharp
```

Comment:
This example doesn't work? You've given an example `DoublingPathParamSerializer`, but the example has enum-like behaviour.

I do think we should probably provide a serializer for enums, either as an example or as actual code (probably as actual code would be better). Then we don't need to deal with `DoublePathParamSerializer` in the examples.

Reply:
Oh yeah, I originally wrote it planning to have some kind of enum serializer but changed my mind and forgot to fix it everywhere I guess.

If we provide one as part of RestEase, should it just look at `EnumMemberAttribute` or also other attributes used for similar purposes like `DisplayAttribute`?

Reply:
I think looking at both EnumMemberAttribute and DisplayAttribute make sense - those are the two that I know about.

One slight complexity is that `EnumMemberAttribute` isn't available in netstandard1.1. We multi-target net45, netstandard1.1 and netstandard2.0, so we'll need some conditional compilation here. I currently only define NETSTANDARD for netstandard1.1 and netstandard2.0, so we'll need something a bit more specific...

`DisplayAttribute` should be available everywhere.

Reply:
I added two new defines `NETSTANDARD1_1` and `NETSTANDARD2_0`. I'm not sure what the serializer should be named though, `EnumRequestPathParamSerializer`? It will only really be useful for enums so maybe that is an appropriate name.

Reply:
Oh actually it looks like all of `NETSTANDARD`, `NETSTANDARD1_1`, and `NETSTANDARD2_0` are automatically defined depending on target framework. So unless there is another reason to explicitly define them it should be safe to remove from the `csproj` and let the build tools define them?

Reply:
> Oh actually it looks like all of NETSTANDARD, NETSTANDARD1_1, and NETSTANDARD2_0 are automatically defined depending on target framework. So unless there is another reason to explicitly define them it should be safe to remove from the csproj and let the build tools define them?

Aha, good find. Sure, take them out of the csproj.

I think `EnumRequestPathParamSerializer` is probably a good name - we won't be able to add any functionality later, so it's always going to be enum-specific. We probably want it to call `ToStringHelper.ToString` on anything that isn't an enum (or any enum member which isn't decorated with one of the attributes), but that means passing in an `IFormatProvider` which none of the serializers currently take... Might have to add that to `RequestPathParamSerializerInfo`.

Reply:
I've added an implementation of a converter in the latest commit. It's not using an `IFormatProvider` though, should that be added somewhere?